### PR TITLE
Add 06_habitat to pipeline and actually push to SB

### PR DESCRIPTION
**Pull request by lindsayplatt**
This PR adds the necessary targets to get 06_habitat data up onto ScienceBase. Once the pipeline was updated, I did actually run that push step to get the two CSV files on SB (🎉). I didn't actually follow the README instructions for doing it from the Yeti directory bc I had already downloaded those large files from Yeti anyways (to send to Gretchen via Google last week) and was blocked by not having write permission to the `.git` folder (@jread-usgs is the only one with write access) . I also have outdate credentials from the cidamanager acct for SB and rather than figuring that all out, I just used my own creds since I was working locally anywasy. You can now see the files on [the habitat SB item](https://www.sciencebase.gov/catalog/item/5db819bbe4b0b0c58b5a4c47)!

### CODE REVIEW

**Comment by lindsayplatt** [comment]
Oh, also worth noting that I plan to update and add to the metadata for 06_habitat in a separate PR (goes with #21)

File: 6_habitat.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+targets:
+  6_habitat:
+    depends:
+      - out_data/6_glm2pb0_annual_metrics.csv
+      - out_data/6_glm2pball_annual_metrics.csv
+
+  out_data/6_glm2pb0_annual_metrics.csv:
+    command: update_hash_path(target_name,
+      '../lake-temperature-out/3_summarize/out/annual_metrics_glm2pb0.csv',
```
Comment (jordansread) [inline 6_habitat.yml]: does calling update_hash_path work here? I thought that function was designed for yaml files??

File: 6_habitat.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      add_path = I('../lake-temperature-out'))
+  
+  out_data/6_glm2pball_annual_metrics.csv:
+    command: update_hash_path(target_name,
```
Comment (jordansread) [inline 6_habitat.yml]: same question here on whether that function works for csvs

File: README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Though we don't strictly adhere to these divisions, here's the general idea for the output folders:
+
+* `out` contains git-committable yml, csv, or ind files that describe the existence of data files
+* `out_data` contains data files that are directly posted to SB as-is
+* `out_xml` contains the rendered XML files that are used to populate SB metadata
+
+## Building 00 through 05
+
+Jordan completes these steps.
```
Comment (jordansread) [inline README.md:14]: Nice README additions!

File: 6_habitat.yml (hunk 1 above)
Comment (lindsayplatt) [reply 6_habitat.yml]: Yes! Works for other files, too. Using it over in mntoha also: https://github.com/USGS-R/mntoha-data-release/blob/master/7_habitat.yml#L26-L29

Comment (lindsayplatt) [reply 6_habitat.yml]: you know though ... even though it works maybe we shouldn't use it and should write something else instead. I am going to update

Comment (lindsayplatt) [reply 6_habitat.yml]: Ok, a simple file.copy is all that is needed here. I changed and rebuilt but the resulting CSVs are the same, so the log file didn't change.

**Comment by jordansread** [comment]
Yeah...but do you remember why file.copy() gave us fits in the slim release template repo?

**Comment by lindsayplatt** [comment]
Oh, no. Did it?

**Comment by lindsayplatt** [comment]
`file.copy` is still part of the README example in slim-data-release. https://github.com/USGS-R/slim-data-release-template/blob/92a9fd3558031395d4849ac53667fd9554db5e1d/README.md

**Comment by jordansread** [comment]
This is the key part: [overwrite = TRUE](https://github.com/USGS-R/slim-data-release-template/blob/main/remake.yml#L30). Otherwise if your `from` file changes but your `to` file already exists, nothing happens and there is no error or message.

**Comment by lindsayplatt** [comment]
That makes sense. I'm glad it is a fairly simple solution to use. I thought you were going to say that it couldn't be used at all. I will try to remember this `gotcha` in the future!

**Pull request merged**